### DISCUSSION TITLE
The website feels way too chaotic

### DISCUSSION THREAD

**Discussion by Weeppiko** [nesbox/TIC-80#2232] [General]
Due to people's lack of common sense (or by being probably kids), it is common to see weird things on the website such as spammy comments or a lot of deleted or duplicates (and probably) stolen carts on the website, with no way of reporting any of these.
The "new & popular" section is simply full of these, completely overshadowing some stuff that could be there instead:
![image](https://github.com/nesbox/TIC-80/assets/88540876/3163d499-abb9-42cd-a61f-1cd1fca1001f)

And I think that the uploader could have some control to at least hide some comments (or stop a user from comment on that specific cart) to prevent stupid spams like this:
![image](https://github.com/nesbox/TIC-80/assets/88540876/e4a59b20-be49-4993-96dc-f5481a52ca6b)

And it really need a way to report users that just post duplicate stuff, like this person that reuploaded my cart without any real significant changes to it (which is the same person as the one who flooded the page with that super mario carts): https://tic80.com/dev?id=8941

**Comment by nesbox** [comment]
@Weeppiko 
I cleaned the DB a little, it's not difficult for me, just write a complaint somewhere.

**Reply by MuubNito** [reply]
Even though it is possible to write a complaint here, I think its more convinient to just add a report button. Not everyone wants to go to a diferrent site (Github) to complain about spam on the website...

**Reply by nesbox** [reply]
I have an idea to move the entire tic80 website to Github and the process of adding carts will be through pull requests.
As a disadvantage, there will be no comments and likes, but now they don't work either, people just spam a lot in comments.

**Reply by nesbox** [reply]
We can use https://github.com/utterance/utterances to add comments and reactions

**Reply by Skeptim** [reply]
Would pull requests be as easy as the current process on the official website? Pull requests (and git in general) have always scared me ^^.

**Reply by sogaiu** [reply]
On a side note, regarding git use, personally, it took me quite some time to become comfortable with using git in the limited way that I do.  I still look stuff up and search for how to do things pretty regularly.  Possibly I feel more at ease now because I think I know better what to avoid and for the tasks I'm faced with, enough know-how has been accumulated (^^;

Compared to when I made my first attempt, I think docs, resources, and software have improved (though now there is more to wade through).  These days there are things like [this](https://gitless.com/) [1] which possibly might work for some folks.

I can totally see not particularly being motivated to learn how to use git though, particulary if not faced with tasks that (nearly) require its use.  Looked at differently, I think there is more reason to learn how to use it if there are multiple things to use it for.

Anyway, just my rapidly devaluing two local currency units :P

---

[1] Apparently there are [a couple of papers](https://gitless.com/#research) that are the basis of / that informed the design of gitless.

**Reply by Skeptim** [reply]
Yes, I agree with you, but it depends on the context. ^^
I have used git a lot for projects I was working on alone and that was a lot of help. But when you have to deal with branches and pull requests, and merging… I had bad experience. But that was very useful here too.
However, since then I left my job (in research) to become a teacher and I don't code any more except on TIC-80 when I have time. So I don't need git any more.
And my question was also because I think TIC-80 is the perfect tool to learn programming therefore the public is not only made of experienced programmers. But after I wrote this message I have looked more at the https://github.com/utterance/utterances link nesbox provided, and I think that the general idea is to use github capabilities in a user friendly manner. That could use pull requests without users knowing they are using pull requests.

**Reply by MuubNito** [reply]
I guess git was designed to be used only by his creator, thats why it is hard (Maybe I am wrong i am talking based on my memory)
